src/Rules/FunctionReturnTypeCheck.php

```diff
@@ -93,7 +94,7 @@ public function checkReturnType(
 			];
 		}
 
-		if (!$this->ruleLevelHelper->accepts($returnType, $returnValueType, $scope->isDeclareStrictTypes())) {
+		if (!$returnValue->hasAttribute(OversizedConstantArrayVisitor::IS_OVERSIZED) && !$this->ruleLevelHelper->accepts($returnType, $returnValueType, $scope->isDeclareStrictTypes())) {
```

Comment:
This isn't a systemic solution. We can't just completely ignore those arrays. Also, there are many other places that check whether Type accepts a Type.

Reply:
Ok I think I need to do this check within the Type classes. 

Do you think I can reuse OversizedArrayType or do I need a completely new one?

Reply:
I've tried to do this (the ConstantArrayType::accepts() change) but it didn't work for me: https://github.com/phpstan/phpstan-src/pull/2077/files#diff-2e28cc3b285eae2b224df51cda5d6d532675e945e2995e2cd76293bca56799d6L303-R309

So maybe you can investigate this path and see what can be done there. The idea is that if an array is dumbed down, it should still be reported if the type is always wrong, like trying to put `array<int>` into `array{foo: string}`.

Reply:
hmm maybe we can `generalize` the accepting type when the accepted type is oversized?

Reply:
ahh nice, inspired by #2077 I guess I found a working solution